x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/apm/traces/unified_trace_root_span.spec.ts

```diff
@@ -0,0 +1,253 @@
+
+export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const apmApiClient = getService('apmApi');
+  const synthtrace = getService('synthtrace');
+
+  const start = new Date('2022-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2022-01-01T00:15:00.000Z').getTime() - 1;
+
+  describe('Unified trace root span', () => {
+    describe('Root span does not exist', () => {
+      it('handles empty state', async () => {
+        const response = await apmApiClient.readUser({
+          endpoint: `GET /internal/apm/unified_traces/{traceId}/root_span`,
+          params: {
+            path: { traceId: 'foo' },
+            query: {
+              start: new Date(start).toISOString(),
+              end: new Date(end).toISOString(),
+            },
+          },
+        });
+
+        expect(response.status).to.be(200);
+        expect(response.body).to.eql({});
+      });
+    });
+
+    describe('when APM root transaction exists', () => {
+      let traceId: string;
+      let apmSynthtraceEsClient: ApmSynthtraceEsClient;
+
+      before(async () => {
+        apmSynthtraceEsClient = await synthtrace.createApmSynthtraceEsClient();
+        const instanceJava = apm
+          .service({ name: 'synth-apple', environment: 'production', agentName: 'java' })
+          .instance('instance-b');
+        const events = timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) => {
+            return [
+              instanceJava
+                .transaction({ transactionName: 'GET /apple 🍏' })
+                .timestamp(timestamp)
+                .duration(1000)
+                .success()
+                .children(
+                  instanceJava
+                    .span({
+                      spanName: 'get_green_apple_🍏',
+                      spanType: 'db',
+                      spanSubtype: 'elasticsearch',
+                    })
+                    .timestamp(timestamp + 50)
+                    .duration(900)
+                    .success()
+                ),
+            ];
+          });
+        const unserialized = Array.from(events);
+
+        const serialized = unserialized.flatMap((event) => event.serialize());
+
+        traceId = serialized[0]['trace.id']!;
+
+        await apmSynthtraceEsClient.index(Readable.from(unserialized));
+      });
+
+      after(() => apmSynthtraceEsClient.clean());
+
+      it('returns APM root transaction duration', async () => {
+        const response = await apmApiClient.readUser({
+          endpoint: `GET /internal/apm/unified_traces/{traceId}/root_span`,
+          params: {
+            path: { traceId },
+            query: {
+              start: new Date(start).toISOString(),
+              end: new Date(end).toISOString(),
+            },
+          },
+        });
+
+        expect(response.status).to.eql(200);
+        expect(response.body).to.not.be(undefined);
```

Comment:
wouldn't be better to explicitly and safer to check for specific fields and their values?